### ISSUE TITLE
Error importing pyw files

### ISSUE THREAD

**Issue by DaSh-More** [microsoft/pylance-release#2332]
## How to reproduce the error
1) Create filse main.py and funcs.pyw in one folder
2) Write the code to the file funcs.pyw:
``` python
def test_func():
    print('hello')
```

2) Write the code to the file main.py:
``` python
from funcs import test_func

test_func()
```

Pylance print:
Import "test9" could not be resolved

---
## Full error info

``` json
[{
	"resource": "/d:/Programs/Python3/test_progs/temp/main.py",
	"owner": "_generated_diagnostic_collection_name_#3",
	"code": {
		"value": "reportMissingImports",
		"target": {
			"$mid": 1,
			"external": "https://github.com/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md#diagnostic-severity-rules",
			"path": "/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md",
			"scheme": "https",
			"authority": "github.com",
			"fragment": "diagnostic-severity-rules"
		}
	},
	"severity": 4,
	"message": "Import \"funcs\" could not be resolved",
	"source": "Pylance",
	"startLineNumber": 1,
	"startColumn": 6,
	"endLineNumber": 1,
	"endColumn": 11
}]
```

_Originally posted by @DaSh-More in https://github.com/microsoft/pylance-release/issues/924#issuecomment-1030199592_

**Comment by MohanElapolu** [comment]
I am facing the same issue in VS code, if two files are in the same folder. It should be able to import from one file to another. Am I missing something?

**Comment by erictraut** [comment]
Is this in the root directory of your project, or is it in a subdirectory? If it's in a subdirectory, you will need to tell pylance that you want to use that subdirectory to resolve imports. You can do this by adding it to the "python.analysis.extraPaths" setting. Does that make the error go away?

**Comment by DaSh-More** [comment]
No, because file in the root directory of project

**Comment by erictraut** [comment]
Ah, I just realized that I misread your original bug report. I read it as "pyd" file, which is a binary library file. I have never heard of "pyw" files before. Pylance doesn't have any knowledge of this file extension. It doesn't attempt to resolve such files currently, so that explains the problem.

**Comment by DaSh-More** [comment]
By default python installs two two interpreter files, python.exe and pythonw.exe . Files with the extension .pyw are launched via pythonw.exe, the only difference of which is that it does not create a separate window, the script is, as it were, executed in the background.

**Comment by erictraut** [comment]
Yes, I see this is a Windows-specific thing. This mechanism must be used rarely because this is the first time this has been reported in the 2+ years that pylance has been under development.

It will take a bit of work to find all of the places where the pylance and pyright code assumes ".py" file extensions and extend it to support ".pyw" as well. This affects not only import resolution but many other pieces of functionality like auto-import suggestions and dynamic mapping between stub and source files.

**Comment by DaSh-More** [comment]
Okay, thanks! I will be waiting for new versions)

**Comment by erictraut** [comment]
Did you mean to close this? I presume you are interested in this feature still? I'll re-open.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
